Shopify/shopify-cli
# Fix partners ability to login to external shops

### WHY are these changes introduced?

Fixes #1833, #1652 (related).
Partners were able to log into stores that were not their own, through `shopify login --store=STORE`, where STORE is not supposed to be accessible by the Partners account.  



### WHAT is this pull request doing?

Previously, the login command was checking the store validity first, then dealing with the user authentication. I changed the code so that it checks that the store belongs to the Partner, after authentication.



### How to test your changes?

1. Find a store that does not belong to your Partners account
2. Run `shopify-dev login --store=STORE`
3. It should fail and show an error message (this is the same message shown when STORE is an invalid store)
** when STORE belongs to the Partners account, login should succeed.



### Post-release steps



### Update checklist
